.github/workflows/test_and_deploy.yml

```diff
@@ -42,12 +42,14 @@ jobs:
         # Run all supported Python versions on linux
         os: [ubuntu-latest]
         python-version: ["3.9", "3.10"]
-        # Include one windows, one macos run
+        # Include one windows, one macos run each for M1 (latest) and Intel (13)
         include:
-        - os: macos-latest
-          python-version: "3.10"
-        - os: windows-latest
-          python-version: "3.10"
+          - os: macos-13
+            python-version: "3.10"
+          - os: macos-latest
+            python-version: "3.10"
+          - os: windows-latest
+            python-version: "3.10"
```

Comment:
question: do you know why we don't need to install `hdf5` here?

Reply:
The `hdf5` installation issue on M1 is linked to depending on `pandas`, which cellfinder doesn't.